pkg/crc/api/api_test.go

```diff
@@ -30,6 +32,21 @@ func newFakeHandler(client *fakemachine.Client) *fakeHandler {
 		}}
 }
 
+// fake Start handler with fake client and without the preflight checks
+func (f *fakeHandler) Start(crcConfig config.Storage, args json.RawMessage) string {
+	parsedArgs, err := parseStartArgs(args)
+	if err != nil {
+		startErr := &machine.StartResult{
+			Name:  "crc",
+			Error: fmt.Sprintf("Incorrect arguments given: %s", err.Error()),
+		}
+		return encodeStructToJSON(startErr)
+	}
+	startConfig := getStartConfig(crcConfig, parsedArgs)
+	status, _ := f.MachineClient.Start(startConfig)
+	return encodeStructToJSON(status)
+}
```

Comment:
What would be great in here would be to have a way to call the normal `api.Start()` handler, with a mock'ed `MachineClient`, but also with mock'ed preflight checks. These fake preflight checks would be just one or 2 checks which would set booleans which we'd use to make sure they were run. I don't know if we can dynamically replace `getPreflightChecks` at runtime ^^ (just food for thoughts, I don't think there is much which can be done right now)